core/src/main/java/org/apache/gravitino/listener/api/info/ModelInfo.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.gravitino.listener.api.info;
+
+import java.util.Collections;
+import java.util.Map;
+import javax.annotation.Nullable;
+import lombok.Getter;
+import org.apache.gravitino.Audit;
+import org.apache.gravitino.annotation.DeveloperApi;
+import org.apache.gravitino.model.Model;
+
+/**
+ * ModelInfo exposes model information for event listener, it's supposed to be read only. Most of
+ * the fields are shallow copied internally not deep copies for performance.
+ */
+@DeveloperApi
+public class ModelInfo {
+  @Getter private final String name;
+  @Nullable private final String comment;
+  @Getter private final Map<String, String> properties;
+  @Nullable private final Audit audit;
+  @Getter private final int lastVersion;
```

Comment:
using `Optional<Integer>` to represent `lastVersion`?  or else it's hard to represent the validation of the field.